pkg/asset/installconfig/azure/dns.go

```diff
@@ -136,7 +156,10 @@ func (client *ZonesClient) GetAllPublicZones() (map[string]string, error) {
 		//the property is present in the REST api response, but not mapped yet in the stable SDK (present in preview)
 		//https://github.com/Azure/azure-sdk-for-go/blob/07f918ba2d513bbc5b75bc4caac845e10f27449e/services/preview/dns/mgmt/2018-03-01-preview/dns/models.go#L857
 		for _, zone := range zonesPage.Values() {
-			allZones[to.String(zone.Name)] = to.String(zone.ID)
+			zoneName := to.String(zone.Name)
+
+			// Should we hash on resource group?
```

Comment:
AFAIK, the zoneID here already has the resource group in it. so zoneDNSName => []{zoneIDs} should be enough to create a map for us.

Reply:
Fixed.